pkg/vulnsrc/oracle-oval/oracle-oval_test.go

```diff
@@ -0,0 +1,873 @@
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			mockDBConfig := new(db.MockDBConfig)
+			mockDBConfig.On("BatchUpdate", mock.Anything).Return(tc.batchUpdateErr)
+			ac := VulnSrc{dbc: mockDBConfig}
+
+			err := ac.Update(tc.cacheDir)
+			switch {
+			case tc.expectedError != nil:
+				assert.EqualError(t, err, tc.expectedError.Error(), tc.name)
+			default:
+				assert.NoError(t, err, tc.name)
+			}
+		})
+	}
+}
+
+func TestVulnSrc_Commit(t *testing.T) {
+	type putAdvisoryInput struct {
+		source   string
+		pkgName  string
+		cveID    string
+		advisory types.Advisory
+	}
+	type putAdvisory struct {
+		input  putAdvisoryInput
+		output error
+	}
+
+	type putVulnerabilityDetailInput struct {
+		cveID  string
+		source string
+		vuln   types.VulnerabilityDetail
+	}
+	type putVulnerabilityDetail struct {
+		input  putVulnerabilityDetailInput
+		output error
+	}
+
+	type putSeverityInput struct {
+		cveID    string
+		severity types.Severity
+	}
+	type putSeverity struct {
+		input  putSeverityInput
+		output error
+	}
+	testCases := []struct {
+		name                       string
+		cves                       []OracleOVAL
+		putAdvisoryList            []putAdvisory
+		putVulnerabilityDetailList []putVulnerabilityDetail
+		putSeverityList            []putSeverity
+		expectedErrorMsg           string
+	}{
+		{
+			name: "happy path",
+			cves: []OracleOVAL{
+				{
+					Title:       "ELSA-2007-0057:  Moderate: bind security update  (MODERATE)",
+					Description: "[30:9.3.3-8]\n - added fix for #224445 - CVE-2007-0493 BIND might crash after\n   attempting to read free()-ed memory\n - added fix for #225229 - CVE-2007-0494 BIND dnssec denial of service\n - Resolves: rhbz#224445\n - Resolves: rhbz#225229",
+					Platform:    []string{"Oracle Linux 5"},
+					References: []Reference{
+						{
+							Source: "elsa",
+							URI:    "http://linux.oracle.com/errata/ELSA-2007-0057.html",
+							ID:     "ELSA-2007-0057",
+						},
+						{
+							Source: "CVE",
+							URI:    "http://linux.oracle.com/cve/CVE-2007-0493.html",
+							ID:     "CVE-2007-0493",
+						},
+						{
+							Source: "CVE",
+							URI:    "http://linux.oracle.com/cve/CVE-2007-0494.html",
+							ID:     "CVE-2007-0494",
+						},
+					},
+					Criteria: Criteria{
+						Operator: "AND",
+						Criterias: []Criteria{
+							{
+								Operator: "OR",
+								Criterias: []Criteria{
+									{
+										Operator:  "AND",
+										Criterias: nil,
+										Criterions: []Criterion{
+											{
+												Comment: "bind-devel is earlier than 30:9.3.3-8.el5",
+											},
+											{
+												Comment: "bind-devel is signed with the Oracle Linux 5 key",
+											},
+										},
+									},
+								},
+								Criterions: nil,
+							},
+						},
+						Criterions: []Criterion{
+							{
+								Comment: "Oracle Linux 5 is installed",
```

Comment:
Would you add a case including two major versions? e.g. Oracle Linux 5 and Oracle Linux 6.